### ISSUE TITLE
The weaver failed to resolve a reference to an assembly - try to use WebPush

### ISSUE THREAD

**Issue by mmnosek** [Starcounter/Starcounter#4182]
I have a separate project with the dll. All of  my apps in the same solutions are dependent on this project. In this project I need `WebPush` from nuget (https://www.nuget.org/packages/WebPush/), so I installed it. Now everything compiles, but I'm unable to run any of my apps. There is an error:

```
The weaver failed to resolve a reference to an assembly. Referenced assembly: System.Net.Http, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a. Probable referrer: Starcounter.Extensions.dll (in C:\Program Files\Starcounter\LibrariesWithDatabaseClasses). (ScErrWeaverFailedResolveReference (SCERR2147))
```

`System.Net.Http` is one of the dependencies of `WebPush` that I've installed. I've tried to add all new dependencies to `weaver.ignore` in my app, so now it looks like this:

```
WebPush.dll
System.Net.Http.dll
BouncyCastle.Crypto.dll
```

It doesn't help. Any idea? What more information I can provide to help you investigating it? I use `Starcounter-RC-2.3.0.5451`

cc @per-samuelsson @chrhol

**Comment by per-samuelsson** [comment]
@mmnosek 

Hard to say right now what is wrong, but my initial guess it's something with (incompatible) versioning of referenced assemblies. That is usually a pain (not only in the weaver specifically, but in .NET in general).

There are 9 earlier issues reporting similar error code. Can you spot something there that could apply to this?

https://github.com/Starcounter/Starcounter/issues?q=is%3Aissue+ScErrWeaverFailedResolveReference

Final help I can offer if you don't find anything is that I get access to the project, set it up myself and see what I can find.

**Comment by per-samuelsson** [comment]
Furthermore, if you have done so already, try to see if **Troubleshooting weaver failures** in https://docs.starcounter.io/guides/working-with-starcounter/weaver/ can help you.

**Comment by mmnosek** [comment]
Per, thanks for the reply! 
One more question - does Starcounter anyhow depend on `System.Net.Http.dll`?

**Comment by per-samuelsson** [comment]
> One more question - does Starcounter anyhow depend on System.Net.Http.dll?

I actually started looking for that, as soon as after I answered. Still haven't found we do, but the error somewhat suggest so.

**Comment by per-samuelsson** [comment]
> I actually started looking for that, as soon as after I answered. Still haven't found we do, but the error somewhat suggest so.

Yeah, looks like `Starcounter.Internal` [actually do](https://github.com/Starcounter/Starcounter/blob/develop/src/Starcounter.Internal/Starcounter.Internal.csproj#L143). 😢

**Comment by mmnosek** [comment]
@per-samuelsson thanks for your investigation! For now, I managed to do some workaround - I've just used WebPush v1.0.6 instead of v1.0.8. It does not depend on `System.Net.Http` for some reason and everything works fine. When I will find some time I am going to isolate this problem and try to describe it better. So I leave this issue open and assign to me.

**Comment by per-samuelsson** [comment]
> When I will find some time I am going to isolate this problem and try to describe it better. So I leave this issue open and **assign to me**.

Thanks a lot. Don't hesitate consulting me if/when you need assistance.
